File: JJB/Programmers/231212/캐릭터의_좌표.py
def solution(keyinputs, board):
    answer = []
    
    xy = [0, 0]
    x_limit = board[0] // 2
    y_limit = board[1] // 2
    
    for ki in keyinputs:
        prev = xy[:]
        
        if ki == "up":
            prev[1] += 1
        elif ki == "down":
            prev[1] -= 1
        elif ki == "left":
            prev[0] -= 1
        elif ki == "right":
            prev[0] += 1
            
        if abs(prev[0]) <= x_limit and abs(prev[1]) <= y_limit:
            xy = prev[:]
            
    answer = xy
    
    return answer

File: JJB/Programmers/231212/캐릭터의_좌표.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> keyinputs, vector<int> board) 
{
    vector<int> answer;
    
    vector<int> xy = {0, 0};
    int x_limit = board[0] / 2;
    int y_limit = board[1] / 2;
    
    for(const string &ki : keyinputs)
    {
        vector<int> prev = xy;
        
        if (ki == "up")
            prev[1]++;
        else if(ki == "down")
            prev[1]--;
        else if (ki == "left")
            prev[0]--;
        else if(ki == "right")
            prev[0]++;
        
        if (abs(prev[0]) <= x_limit && abs(prev[1]) <= y_limit)
            xy = prev;
    }
    
    answer = xy;
    
    return answer;
}